### PR TITLE
feat: Add Support for net.IP in Faker

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

The Gitlab Plugin I am working on has `net.IP` fields... existing implementation does not produce a valid IP address so needs explicit support

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @bbernays, added 2 comments.
Can you also link to which mock test faking an IP is required?

File: faker/faker.go [hunk 1]
```diff
@@ -70,17 +69,29 @@ func (f faker) getFakedValue(a interface{}) (reflect.Value, error) {
...
 		return reflect.ValueOf("test string"), nil
 	case reflect.Slice:
-		sliceLen := 1
-		v := reflect.MakeSlice(t, sliceLen, sliceLen)
-		for i := 0; i < v.Len(); i++ {
-			val, err := f.getFakedValue(v.Index(i).Interface())
-			if err != nil {
-				return reflect.Value{}, err
+		switch t.String() {
+		case "net.IP":
```
[the remark this answers is not in the document]
Comment (bbernays) [reply faker/faker.go:73]: This is how `time.Time` is handled on line 46... Happy to change, but was trying to keep consistency within the function

File: faker/faker_test.go [hunk 2]
```diff
@@ -46,3 +47,16 @@ func TestFakerWithCustomType(t *testing.T) {
...
+type complexType struct {
+	F net.IP
+}
+
+func TestFakerWithComplexCustomType(t *testing.T) {
+	a := complexType{}
+	if err := FakeObject(&a); err != nil {
+		t.Fatal(err)
+	}
+	assert.NotEmpty(t, a.F)
```
[the remark this answers is not in the document]
Comment (bbernays) [reply faker/faker_test.go:60]: Yeah will do that. Haven't finished the tests yet

File: faker/faker.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (erezrokah) [reply faker/faker.go:73]: Got, so we can keep it like this and refactor in a future PR

File: faker/faker_test.go [hunk 3]
```diff
@@ -46,3 +47,16 @@ func TestFakerWithCustomType(t *testing.T) {
 	assert.NotEmpty(t, a.C)
 	assert.NotEmpty(t, a.D)
 }
+
+type complexType struct {
+	IPAddress net.IP
+}
+
+func TestFakerWithComplexCustomType(t *testing.T) {
```
Comment (erezrokah) [inline faker/faker_test.go:55]:
```suggestion
func TestFakerCanFakeNet.IP(t *testing.T) {
```

[non blocking] Maybe a name that describes what we're testing in a more specific way

**Review by yevgenypats — CHANGES_REQUESTED** [review]
doesn't that produce a valid IP only for for slice? what about `map`, `ptr` or just field in struct ?

**Comment by yevgenypats** [comment]
Or maybe I miss something as seems the tests does pass? didn't touch that code for a while....

**Comment by bbernays** [comment]
> doesn't that produce a valid IP only for for slice? what about map, ptr or just field in struct ? 

`net.IP` is defined as as a `type IP []byte` slice so we only have to handle that one special type of `slice`

**Pull request merged**